x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/notifications/AbstractAuditor.java

```diff
@@ -64,16 +100,65 @@ private void onIndexFailure(Exception exception) {
     }
 
     private void indexDoc(ToXContent toXContent) {
+        if (hasLatestTemplate.get()) {
+            writeDoc(toXContent);
+            return;
+        }
+
+        if (MlIndexAndAlias.hasIndexTemplate(clusterService.state(), templateName)) {
+            synchronized (this) {
```

Comment:
This synchronized block seems unnecessary.

Reply:
It is because of the code at line 131
```
        synchronized (this) {
            if (hasLatestTemplate.get() == false) {
                // synchronized so that hasLatestTemplate does not change value
                // between the read and adding to the backlog
                backlog.add(toXContent);
```

I need to make reading `hasLatestTemplate` and adding to `backlog` an atomic operation otherwise `hasLatestTemplate` could change value between the 2 lines of code. 

Nothing can be added to `backlog` once `hasLatestTemplate` is true otherwise it may not be written in `writeBacklog`